mteb/models/jina_models.py

```diff
@@ -0,0 +1,127 @@
+            )
+        super().__init__(model, revision, model_prompts, **kwargs)
+
+    def encode(
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        """Encodes the given sentences using the encoder.
+
+        Args:
+            sentences: The sentences to encode.
+            task_name: The name of the task. Sentence-transformers uses this to
+                determine which prompt to use from a specified dictionary.
+            prompt_type: The name type of prompt. (query or passage)
+            **kwargs: Additional arguments to pass to the encoder.
+
+            The order of priorities for prompt selection are:
+                1. Composed prompt of task name + prompt type (query or passage)
+                2. Specific task prompt
+                3. Composed prompt of task type + prompt type (query or passage)
+                4. Specific task type prompt
+                5. Specific prompt type (query or passage)
+
+
+        Returns:
+            The encoded sentences.
+        """
+        prompt_name = get_prompt_name(self.model_prompts, task_name, prompt_type)
+        if prompt_name:
+            logger.info(
+                f"Using prompt_nane={prompt_name} for task={task_name} prompt_type={prompt_type}"
+            )
+        else:
+            logger.info(
+                f"No model prompts found for task={task_name} prompt_type={prompt_type}"
+            )
+        logger.info(f"Encoding {len(sentences)} sentences.")
+
+        jina_task_name = self.model_prompts.get(prompt_name, None)
+
+        embeddings = self.model.encode(
+            sentences,
+            task=jina_task_name,
+            prompt=self.jina_task_to_prompt.get(jina_task_name, None),
+            **kwargs,
+        )
+
+        if isinstance(embeddings, torch.Tensor):
+            # sometimes in kwargs can be return_tensors=True
+            embeddings = embeddings.cpu().detach().float().numpy()
+        return embeddings
+
+
+jina_embeddings_v3 = ModelMeta(
+    loader=partial(
+        JinaWrapper,
+        model="jinaai/jina-embeddings-v3",
+        revision="215a6e121fa0183376388ac6b1ae230326bfeaed",
+        model_prompts={
+            "Retrieval-query": "retrieval.query",
+            "Retrieval-passage": "retrieval.passage",
+            "Clustering": "separation",
+            "Classification": "classification",
+            "STS": "text-matching",
+            "PairClassification": "classification",
+            "BitextMining": "text-matching",
+            "MultilabelClassification": "classification",
+            "Reranking": "separation",
+            "Summarization": "text-matching",
+        },
+    ),
+    name="jinaai/jina-embeddings-v3",
+    languages=["eng_Latn"],
+    open_source=True,
```

Comment:
maybe it's nice to add a small comment, making sure ppl aware that it's an open-weights model, for example:
```suggestion
    open_source=True, # cc-by-nc-4.0
```

Reply:
There is addional field in modelmetdata. I will update metadata a bit later